### PR TITLE
3798 utility gateway/chris wagner

### PR DESCRIPTION
**Pull request by Cwagne17**
PR completes [Story 3798](https://app.clubhouse.io/clarkcan/story/3798/group-utility-service-routes-into-a-controller).

The PR moves over the Utility Service routes into the UtilityServiceController
- All routes work properly still
- Documentation was added in swagger

### CODE REVIEW

File: src/modules/utility-service/UtilityServiceController.ts [hunk 1]
```diff
@@ -1,12 +1,115 @@
...
+         *          200:
+         *              description: OK - Returns status report
+         */
+        router.get(
+            '/status',
+            proxy(UTILITY_API, {
+                proxyReqPathResolver: req => {
+                    return UTILITY_ROUTES.STATUS;
+                },
+            }),
```
Comment (EVogel1999) [inline src/modules/utility-service/UtilityServiceController.ts:29]: Let's try to follow the pattern set up in the other controllers where we pass in a callback to the proxy request function.

**Review by EVogel1999 — CHANGES_REQUESTED** [review]
If things are being returned such as a status report then we should be describing the object being returned or sent, not just saying it returns something.  If these objects are complex you can define them in a Component.ts file at the same level as the controller.  You shouldn't be able to test /outages locally (because we can't replicate the database locally) though so you'll have to go to prod to know what is being returned from that.  Otherwise it looks good!

**Review by pzalep1 — COMMENTED** [review]
After Emily's comments are addressed this looks good!

File: src/modules/utility-service/UtilityServiceController.ts [hunk 2]
```diff
@@ -1,12 +1,121 @@
...
+         *                          items:
+         *                              $ref: '#/components/schemas/Outage'
+         *                          
+         *                      
+         *          500:
+         *              description: INTERNAL - Unable to get system outages
+         */
+        router.get(
+            '/outages',
+            //this.proxyRequest((req: Request) => `/outages?pastIssues=${encodeURIComponent(req.query.pastIssues)}`)
```
Comment (EVogel1999) [inline src/modules/utility-service/UtilityServiceController.ts:109]: Uncomment this.

File: src/modules/utility-service/UtilityServiceController.ts [hunk 3]
```diff
@@ -1,12 +1,121 @@
...
+         *          401:
+         *              description: UNAUTHENTICATED - Thrown when no clientVersion is in params
+         *          426:
+         *              description: Thrown when the client version is not accurate
+         *          500: 
+         *              description: INTERNAL - Could not recover the client version
+         */
+        router.get(
+            '/clientversion/:clientVersion',
+            //this.proxyRequest((req: Request) => `/clientversion/${encodeURIComponent(req.params.clientVersion)}`)
```
Comment (EVogel1999) [inline src/modules/utility-service/UtilityServiceController.ts:76]: Uncomment this.

File: src/modules/utility-service/UtilityServiceController.ts [hunk 4]
```diff
@@ -1,12 +1,121 @@
...
+         *            name: clientVersion
+         *            schema:
+         *                type: string
+         *            required: true
+         *            description: The version of the client
+         *      responses:
+         *          200:
+         *              description: OK
+         *          401:
+         *              description: UNAUTHENTICATED - Thrown when no clientVersion is in params
```
Comment (EVogel1999) [inline src/modules/utility-service/UtilityServiceController.ts:68]: Quick question is this supposed to be a 401?  This seems more like a 400 error but we may have incorrectly mapped it

**Review by EVogel1999 — COMMENTED** [review]
Just a few minor things then it looks good to go!

File: src/modules/utility-service/UtilityServiceController.ts (hunk 4 above)
Comment (Cwagne17) [reply src/modules/utility-service/UtilityServiceController.ts:68]: It was returning a 401, I do agree a 400 seems more appropriate

**Review by pzalep1 — APPROVED** [review]
Looks good!

**Review by jmulholland4014 — APPROVED** [review]

**Review by mfranz1 — APPROVED** [review]

**Review by devky690 — APPROVED** [review]

**Review by EVogel1999 — APPROVED** [review]

**Pull request merged**